blockmanager.go

```diff
@@ -700,17 +698,16 @@ func (b *blockManager) checkBlockForHiddenVotes(block *dcrutil.Block) {
 	// the parent template hasn't yet been updated, so we may
 	// need to use the current template.
 	var template *BlockTemplate
-	if b.cachedCurrentTemplate != nil {
-		if b.cachedCurrentTemplate.Height ==
-			block.Height() {
-			template = b.cachedCurrentTemplate
+	if b.server.bg != nil {
+		if b.server.bg.currentTemplate != nil {
```

Comment:
Race.  `currentTemplate` is supposed to protected by `templateMtx`.  There are several more cases, but I won't comment on them.